ddtrace/contrib/aiomysql/patch.py

```diff
@@ -0,0 +1,118 @@
+import asyncio
+
+import aiomysql
+
+from ddtrace import Pin
+from ddtrace import config
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.constants import SPAN_MEASURED_KEY
+from ddtrace.contrib import dbapi
+from ddtrace.ext import sql
+from ddtrace.internal.utils.wrappers import unwrap
+from ddtrace.vendor import wrapt
+
+from ...ext import SpanTypes
+from ...ext import db
+from ...ext import net
+
+
+config._add(
+    "aiomysql",
+    dict(_default_service="mysql"),
+)
+
+CONN_ATTR_BY_TAG = {
+    net.TARGET_HOST: "host",
+    net.TARGET_PORT: "port",
+    db.USER: "user",
+    db.NAME: "db",
+}
+
+
+async def patched_connect(connect_func, _, args, kwargs):
+    conn = await connect_func(*args, **kwargs)
+    tags = {t: getattr(conn, a, "") for t, a in CONN_ATTR_BY_TAG.items()}
+
+    c = AIOTracedConnection(conn)
+    Pin(tags=tags).onto(c)
+    return c
+
+
+class AIOTracedCursor(wrapt.ObjectProxy):
+    """TracedCursor wraps a aiomysql cursor and traces its queries."""
+
+    def __init__(self, cursor, pin):
+        super(AIOTracedCursor, self).__init__(cursor)
+        pin.onto(self)
+        self._datadog_name = "mysql.query"
```

Comment:
```suggestion
        self._self_datadog_name = "mysql.query"
```

`wrapt.ObjectProxy` is weird, in order to set an attribute on the `ObjectProxy` itself instead of proxying that `setattr()` to the wrapped instance, you need to prefix the attribute with `_self_<name>`.

this is probably working, since the wrapped instance allows you to set an arbitrary attribute, but if the wrapped instance had `__slots__` defined, then it could cause issues.